build.gradle

```diff
@@ -110,3 +110,14 @@ testClusters.integTest {
 run {
     useCluster testClusters.integTest
 }
+
+// versionIncrement: Task to auto increment to the next development iteration
+task versionIncrement {
+    onlyIf { System.getProperty('newVersion') }
+    doLast {
+        ext.newVersion = System.getProperty('newVersion')
+        println "Setting version to ${newVersion}."
+         // String tokenization to support -SNAPSHOT
+        ant.replaceregexp(file:'build.gradle', match: '"opensearch.version", "\\d.*"', replace: '"opensearch.version", "' + newVersion.tokenize('-')[0] + '-SNAPSHOT"', flags:'g', byline:true)
+    }
+}
```

Comment:
More like `setVersion`?

Reply:
@prudhvigodithi All @AmiStrn is saying is that the task does one thing (set a version), but is used differently (increment the version). You should name the task to match what it does, not how it's used.

Reply:
Im sorry i was not clear. 
The issue is precisely what @dblock understood.
You can potentially set to version 1.0 right? so it is not "increment", `setVersion` would be more adequate, maybe `updateVersion`?

Reply:
Thanks @AmiStrn,
The idea is to call this task via [workflow](https://github.com/opensearch-project/opensearch-build/pull/2291), the workflow only compares with OpenSearch and sets the version, so it would be an increment as OpenSearch core version would be always ahead and with existing setup, based on OpenSearch core version plugin versions are inferred. I'm good with `updateVersion`.